test/test_autograd.py

```diff
@@ -3965,6 +3955,131 @@ def backward(ctx, grad_out):
         gradcheck(lambda x: NonDetFunc.apply(x, 1e-6), inp, nondet_tol=1e-5, check_batched_grad=False)
         gradgradcheck(lambda x: NonDetFunc.apply(x, 1e-12), inp, nondet_tol=1e-5, check_batched_grad=False)
 
+    def test_gradcheck_test_inputs(self):
+        # when inputs are not dense, but check_sparse_nnz is false
+        x = torch.rand(10, requires_grad=True).to_sparse()
+        with self.assertRaisesRegex(RuntimeError, 'dense when check_sparse_nnz is set to False.'):
+            gradcheck(lambda x: x.to_dense(), (x,), check_sparse_nnz=False, check_batched_grad=False)
```

Comment:
Should we be careful that we're testing the public gradcheck API here (torch.autograd.gradcheck)? I think right now we're using the private gradcheck API.

For the purposes of this test it looks like it doesn't actually matter, but if we diverge the behavior of the public and testing-only gradcheck more then it may matter.

Reply:
Good point. we can revisit this when we start to think about how we test that slow/fast have the same behavior.